### DISCUSSION TITLE
Need for additional button variant: "Text"

### DISCUSSION THREAD

**Discussion by janhassel** [carbon-design-system/carbon#14252] [Ideas]
I've observed the following situation in many designs over the last years and think it would be worth to create a new button variant for this.

**Scenario:**
There is a need for a button that aligns with body text and doesn't take up as much space as the ghost button. In these cases, I've often seen the Link component being used to achieve the appropriate visual output. This however creates misleading semantics for users with assistive technologies as these links are not actually navigational but perform an action.

**Example:**
"See more" button to expand and collapse the full text.

*Left: Link, misleading semantics but better visual result
Middle: Small ghost button, correct semantics but awkward visual result
Right: Same as middle, just shifted 1rem to the left. Awkward visual result when hovered or focused*

![image](https://github.com/carbon-design-system/carbon/assets/28265588/a6481313-37a4-408c-af57-25d48a3b13d4)

---

Creating a new button variant "Text" (not set on the naming) would help improve accessibility in these use cases (I can share more via slack if needed). It would ideally have no padding to sit flush with any adjacent elements. The hover and focus states would need to be considered (should they mimic the link states or still have more of a button-feeling?)

**Comment by AmandaFurukawa** [comment]
some more example scenarios:

select all for a group of checkboxes (deselect all)
<img width="294" alt="image" src="https://github.com/carbon-design-system/carbon/assets/119629745/7436a364-8510-4248-afd5-9ac9ceaa0453">

view more
<img width="241" alt="image" src="https://github.com/carbon-design-system/carbon/assets/119629745/5363175a-ba4e-4dc6-b2b4-3146397eb256">

this was implemented incorrect - it opens a translate popover
<img width="643" alt="image" src="https://github.com/carbon-design-system/carbon/assets/119629745/7b38f01b-cc75-4c91-84db-c583b4bc369b">

**Comment by ConradSchmidt** [comment]
I have a similar problem in the (IBM) product I'm working for. As @janhassel was mentioning, the `Link` alternative is not matching from the interaction pattern of "calling" something rather than opening a link. The link only accepts a `href` which I don't need in this case, as I want to call a JS function. If I use the Link with the `onClick` attribute, I get these problems:
1. can't be focused (e.g. using `Tab`, because of the missing `href`)
2. can't be triggered using `Enter` or `Space`(I have to write a separate handler)

Side notes:
If I add `href="javascript:void(0)` to get the element focused, I get warnings as this is not good practice. What I saw is that devs are using `href="-1"` (or similar) and then you have to `preventDefault` which is just more unnecessary code (which needs to be maintained).

Point 2. is the same with the `ClickableTile`, which is also rendered as a `a`nchor tag. Not sure why its using this tag, but I have to specifically implement the `onKeyDown` to have proper accessibility.

Here is an image of the used Button like Link ("View all"):
![image](https://github.com/user-attachments/assets/8cb0b1cc-e844-448a-af36-9d93752784ee)

**Comment by devadula-nandan** [comment]
we have a similar issue here as well in ibm-products where an inline button variant could be helpful as mentioned above
https://ibm-products-web-components.carbondesignsystem.com/?path=/story/utilities-truncatedtext--type-expand
